types/package.json

```diff
@@ -1,13 +1,29 @@
 {
   "name": "@joystream/types",
-  "version": "0.20.3",
+  "version": "0.20.4",
   "description": "Types for Joystream Substrate Runtime - Carthage release",
-  "main": "index.js",
-  "types": "index.d.ts",
+  "main": "lib/index.js",
```

Comment:
Or am I testing it incorrectly?

Reply:
seems to be fine if I create a simple npm package in the workspace and consume the types package.

Reply:
Yes, you are right the error only occurs in vanilla javascript projects. 

`@joystream/types` package.json has an `exports` field.

```json
 "exports": {
    ".": {
      "import": "./lib/mjs/index.js",
      "require": "./lib/cjs/index.js"
    }
```
Node documentation states that you can use both `"main"` & `"exports"` as module [entrypoints](https://nodejs.org/api/packages.html#package-entry-points). Also, from the docs,

> For packages supporting Node.js 10 and below, the ["main"](https://nodejs.org/api/packages.html#main) field is required. If both ["exports"](https://nodejs.org/api/packages.html#exports) and ["main"](https://nodejs.org/api/packages.html#main) are defined, the ["exports"](https://nodejs.org/api/packages.html#exports) field takes precedence over ["main"](https://nodejs.org/api/packages.html#main) in supported versions of Node.js.

In fact, "main" is redundant when "exports" is used. so I tried removing `"main"` field, so that vanilla node project would resolve the module using the "exports" entry, but it wasn't the case, and I got the error. No sure what's causing this issue.

```
Uncaught Error: Cannot find module './'
Require stack:
- <repl>
    at Module._resolveFilename (node:internal/modules/cjs/loader:995:15)
    at Module._load (node:internal/modules/cjs/loader:841:27)
    at Module.require (node:internal/modules/cjs/loader:1061:19)
    at require (node:internal/modules/cjs/helpers:103:18) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [ '<repl>' ]
}
```

Anyways I have updated the main entry as requested in [8e05b04](https://github.com/Joystream/joystream/pull/4435/commits/8e05b04b84239b34db9f442ccfe9b907b2aa0d20)